memiavl/export.go

```diff
@@ -0,0 +1,31 @@
+package memiavl
+
+import (
+	"github.com/cosmos/iavl"
+)
+
+type Exporter struct {
+	snapshot *Snapshot
+	i        uint32
+	count    int
+}
+
+func (e *Exporter) Next() (*iavl.ExportNode, error) {
+	if int(e.i) >= e.count {
+		return nil, iavl.ExportDone
+	}
+	node := e.snapshot.Node(e.i)
+	e.i++
```

Comment:
will e.i++ dirty if concurrent?

Reply:
it's not thread-safe

Reply:
so we don't need function lock?

Reply:
yes, it's always [used in a sequential way](https://github.com/cosmos/cosmos-sdk/blob/main/store/rootmulti/store.go#L781), there's no point a do an iteration concurrently, the result is un-deterministic.